### PR TITLE
feat(csam): add toggle for csam scan and save info to textile

### PR DESCRIPTION
**Pull request by genie-magic**
**What this PR does** 📖
- This is rebased PR based on https://github.com/Satellite-im/Core-PWA/pull/2780
- add a toggle for can scan and save info to textile
- add textile manager for saving user consent data

**Which issue(s) this PR fixes** 🔨

AP-1332

**Special notes for reviewers** 🗒️
- This is already reviewed by devs on old PR... This is just new PR with rebase dev.

### CODE REVIEW

**Review by phillsatellite — APPROVED** [review]

**Review by stavares843 — APPROVED** [review]

File: types/textile/user.ts [hunk 1]
```diff
@@ -0,0 +1,7 @@
+export interface UserdataFromThread {
+  _id: string
+  created_at: Number
```
Comment (WanderingHogan) [inline types/textile/user.ts:3]: can you make these lowercase

Comment (genie-magic) [reply types/textile/user.ts:3]: you mean UserdataFromThread @WanderingHogan ?

Comment (genie-magic) [reply types/textile/user.ts]: @WanderingHogan  I changed type as lowercase.

File: libraries/Textile/UserManager.ts [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+  /**
+   * @method setConsent
+   * Set consent data for csam
+   */
+  async setConsent({
+    consentScan,
+    consentDate,
+  }: {
+    consentScan: boolean
+    consentDate: Number
```
Comment (stavares843) [inline libraries/Textile/UserManager.ts:78]: does it need to be changed here as well? cc @WanderingHogan

Comment (genie-magic) [reply libraries/Textile/UserManager.ts]: @stavares843 That issue is fixed too. thanks for your information.

**Review by WanderingHogan — APPROVED** [review]

**Review by stavares843 — APPROVED** [review]

**Pull request merged**